### PR TITLE
LPD-39485 [HL] Investigate failures in headless-builder-web/schemas.spec.ts        

### PR DESCRIPTION
**Pull request by magjed4289**
https://issues.liferay.com/browse/LPD-39485

### CODE REVIEW

**Comment by magjed4289** [comment]
ci:test:headless-playwright

**Comment by magjed4289** [comment]
ci:test:headless-playwright

**Comment by magjed4289** [comment]
ci:test:headless-playwright

**Comment by magjed4289** [comment]
ci:test:headless-playwright

**Comment by magjed4289** [comment]
ci:test:sf

**Comment by magjed4289** [comment]
ci:test:sf

**Comment by magjed4289** [comment]
ci:test:sf

**Comment by magjed4289** [comment]
ci:test:sf

File: modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts [hunk 1]
```diff
@@ -48,9 +51,8 @@ const singleElementIdEndpoint = {
 test('can see filter and sort parameters for collection endpoints', async ({
 	apiExplorerPage,
 	apiHelpers,
-	page,
 }) => {
-	await apiHelpers.objectEntry.postObjectEntry(
+	const collectionEndpoint = await apiHelpers.objectEntry.postObjectEntry(
```
Comment (ccorreagg) [inline modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts:55]: This is an application, as you are creating an application with the application data and with an endpoint. Please rename it to something with more sense, like `const apiApplication'

File: modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts [hunk 2]
```diff
@@ -68,26 +70,21 @@ test('can see filter and sort parameters for collection endpoints', async ({
...
-	);
 });
 
 test('can see get endpoint path with erc parameter', async ({
 	apiExplorerPage,
 	apiHelpers,
-	page,
 }) => {
-	await apiHelpers.objectEntry.postObjectEntry(
+	const singleElementEndpoint = await apiHelpers.objectEntry.postObjectEntry(
```
Comment (ccorreagg) [inline modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts:87]: Same here

File: modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts [hunk 3]
```diff
@@ -118,27 +120,24 @@ test('can see get endpoint path with erc parameter', async ({
...
 	apiExplorerPage,
 	apiHelpers,
-	page,
 }) => {
-	await apiHelpers.objectEntry.postObjectEntry(
-		{
-			...application,
-			apiApplicationToAPIEndpoints: [singleElementIdEndpoint],
-		},
+	const applicationEntry = await apiHelpers.objectEntry.postObjectEntry(
```
Comment (ccorreagg) [inline modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts:129]: Same here

File: modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts [hunk 4]
```diff
@@ -149,37 +148,28 @@ test('can see get endpoint path with id parameter', async ({
...
```
Comment (ccorreagg) [inline modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts:157]: Same here. Unify

File: modules/test/playwright/tests/headless-builder-web/get.spec.ts [hunk 5]
```diff
@@ -118,6 +120,14 @@ test('can see available path parameter properties of a singleElement endpoint',
 	await headlessBuilderPage.newApplicationTitleBox.fill('My-app');
 	await headlessBuilderPage.createApplicationButton.click();
 
+	const apiApplicationPage =
```
Comment (ccorreagg) [inline modules/test/playwright/tests/headless-builder-web/get.spec.ts:123]: In the line 98 of this file I still see:

```
await apiHelpers.objectEntry.deleteObjectEntryByExternalReferenceCode(
```

Is that still necessary?

Comment (ccorreagg) [reply modules/test/playwright/tests/headless-builder-web/get.spec.ts:123]: and in the line 252

**Review by ccorreagg — COMMENTED** [review]
Please review my comments

File: modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts (hunk 1 above)
Comment (magjed4289) [reply modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts]: see https://github.com/liferay-headless/liferay-portal/pull/2322/commits/8c36851645a50ce9f9faf7ceb1445fb98970bea7

File: modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts (hunk 3 above)
Comment (magjed4289) [reply modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts]: see https://github.com/liferay-headless/liferay-portal/pull/2322/commits/8c36851645a50ce9f9faf7ceb1445fb98970bea7

File: modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts (hunk 2 above)
Comment (magjed4289) [reply modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts]: see https://github.com/liferay-headless/liferay-portal/pull/2322/commits/8c36851645a50ce9f9faf7ceb1445fb98970bea7

File: modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts (hunk 4 above)
Comment (magjed4289) [reply modules/test/playwright/tests/headless-builder-web/apiExplorer.spec.ts]: see https://github.com/liferay-headless/liferay-portal/pull/2322/commits/8c36851645a50ce9f9faf7ceb1445fb98970bea7

File: modules/test/playwright/tests/headless-builder-web/get.spec.ts (hunk 5 above)
Comment (magjed4289) [reply modules/test/playwright/tests/headless-builder-web/get.spec.ts:118]: see https://github.com/liferay-headless/liferay-portal/pull/2322/commits/952ebba82152c69dc3f1ee4a6ecf90c0ca241ede

**Comment by magjed4289** [comment]
ci:test:headless-playwright:nocompile

**Comment by magjed4289** [comment]
ci:test:sf

**Comment by magjed4289** [comment]
ci:test:headless-playwright:nocompile

**Comment by magjed4289** [comment]
ci:test:headless-playwright

**Review by ccorreagg — APPROVED** [review]
LGTM.

As we modify the ApiHelpers, maybe it would be good to execute all the playwright functional tests to check we are not breaking anything with this PR before forwarding

**Comment by ccorreagg** [comment]
ci:help

**Comment by ccorreagg** [comment]
ci:test:playwright

**Comment by magjed4289** [comment]
ci:forward

**Pull request closed without merging**